evm_loader/lib/src/commands/emulate.rs

```diff
@@ -97,18 +99,22 @@ async fn emulate_trx<T: Tracer>(
     let chain_id = tx.chain_id().unwrap_or_else(|| storage.default_chain_id());
     storage.use_balance_account(origin, chain_id, true).await?;
 
-    let mut backend = ExecutorState::new(storage);
-    let mut evm = match Machine::new(&tx, origin, &mut backend, tracer).await {
-        Ok(evm) => evm,
-        Err(e) => return Ok((EmulateResponse::revert(e), None)),
-    };
-
-    let (exit_status, steps_executed, tracer) = evm.execute(step_limit, &mut backend).await?;
-    if exit_status == ExitStatus::StepLimit {
-        return Err(NeonError::TooManySteps);
-    }
+    // Execute and return results to restrict the lifetime of mutable borrow.
+    let (actions, exit_status, steps_executed, tracer) = {
+        let mut executor_state_data = ExecutorStateData::new(storage);
+        let mut backend = ExecutorState::new(storage, &mut executor_state_data);
+        let mut evm = match Machine::new(&tx, origin, &mut backend, tracer).await {
+            Ok(evm) => evm,
+            Err(e) => return Ok((EmulateResponse::revert(e), None)),
+        };
+
+        let (exit_status, steps_executed, tracer) = evm.execute(step_limit, &mut backend).await?;
+        if exit_status == ExitStatus::StepLimit {
+            return Err(NeonError::TooManySteps);
+        }
 
-    let actions = backend.into_actions();
+        (backend.into_actions(), exit_status, steps_executed, tracer)
+    };
```

Comment:
I suggest only reducing the scope of `let mut evm` and the mutable borrow needed by it in `evm.execute(step_limit, &mut backend).await?`, so the block can be even shorter:
```Rust
    let (exit_status, steps_executed, tracer) = {
        let mut evm = match Machine::new(&tx, origin, &mut backend, tracer).await {
            Ok(evm) => evm,
            Err(e) => return Ok((EmulateResponse::revert(e), None)),
        };

        evm.execute(step_limit, &mut backend).await?
    };
```

Reply:
thanks, done.